src/main/java/com/google/api/codegen/util/nodejs/NodeJSTypeTable.java

```diff
@@ -14,45 +14,36 @@
  */
 package com.google.api.codegen.util.nodejs;
 
-import java.util.Map;
-import java.util.TreeMap;
-
+import com.google.api.codegen.util.DynamicLangTypeTable;
 import com.google.api.codegen.util.NamePath;
 import com.google.api.codegen.util.TypeAlias;
 import com.google.api.codegen.util.TypeName;
 import com.google.api.codegen.util.TypeTable;
-import com.google.common.collect.BiMap;
-import com.google.common.collect.HashBiMap;
 import com.google.common.collect.ImmutableSet;
+import java.util.Map;
 
-/**
- * The TypeTable for NodeJS.
- */
+/** The TypeTable for NodeJS. */
 public class NodeJSTypeTable implements TypeTable {
-  /**
-   * A bi-map from full names to short names indicating the import map.
-   */
-  private final BiMap<String, String> imports = HashBiMap.create();
 
-  private final String implicitPackageName;
+  private final DynamicLangTypeTable dynamicTypeTable;
```

Comment:
I won't block the review on this issue, but since the dynamic language type tables have more in common than they are different, I would have expected the composition to be arranged differently, where DynamicLangTypeTable implemented the TypeTable interface, and then had a strategy object with language-specific behavior.